test/builder/container.go

```diff
@@ -67,3 +74,45 @@ func VolumeMount(name, mountPath string, ops ...VolumeMountOp) ContainerOp {
 		c.VolumeMounts = append(c.VolumeMounts, *mount)
 	}
 }
+
+func Resources(ops ...ResourceRequirementsOp) ContainerOp {
+	return func(c *corev1.Container) {
+		rr := &corev1.ResourceRequirements{}
+		for _, op := range ops {
+			op(rr)
+		}
+		c.Resources = *rr
+	}
+}
+
+func Limits(ops ...ResourceListOp) ResourceRequirementsOp {
+	return func(rr *corev1.ResourceRequirements) {
+		limits := corev1.ResourceList{}
+		for _, op := range ops {
+			op(limits)
+		}
+		rr.Limits = limits
+	}
+}
+
+func Requests(ops ...ResourceListOp) ResourceRequirementsOp {
+	return func(rr *corev1.ResourceRequirements) {
+		requests := corev1.ResourceList{}
+		for _, op := range ops {
+			op(requests)
+		}
+		rr.Requests = requests
+	}
+}
+
+func CPU(val string) ResourceListOp {
+	return func(r corev1.ResourceList) {
+		r[corev1.ResourceCPU] = resource.MustParse(val)
+	}
+}
+
+func Memory(val string) ResourceListOp {
```

Comment:
Alternatively we could just have `func Resource(name corev1.ResourceName, val string)` so we don't need a separate method for every function, but that is a little more verbose.